006-strimzi-ui.md

```diff
@@ -0,0 +1,120 @@
+
+### UI implementation
+
+At a high level, I would propose a UI implemented as follows:
+
+- It would be a Javascript based UI (using [babel](https://babeljs.io/) to provide latest ECMAScript capabilities in a cross browser compatible manner), using the [React](https://reactjs.org/) framework
+- This UI is hosted and provided via an [Express](https://expressjs.com/) server
+- This UI is built in a modular/metadata driven manner - allowing for easy extensibility, modification, and dynamic behaviour at runtime
+- Uses a [GraphQL](https://graphql.org/) API
+- Should be fully translatable, and accessible
+
+I would suggest that the UI implementation to follow the [Model View Controller](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller) (MVC) pattern, with clear separation between business logic, and the view logic that renders the state of the UI. This would enable/allow the view logic to be modified/swapped in/out as required, but keep the same business logic. I would suggest using the [Carbon](https://www.carbondesignsystem.com/) design system as the view layer for the contributed UI, given it is open source, is supported, and I can provide both the design and implementation using it. As an example, the following is a mock up of how a Topics page could look using the Carbon design system, which would allow a user to view, create, edit and delete topics in a given cluster:
+
+![Topics mock up](./006-topicsdesign.png)
+
+By maintaining and enforcing a thin view layer, this should not stop other view layers being designed, implemented and contributed in other frameworks, such as [PatternFly](https://www.patternfly.org/v4/) or [Material](https://material.io/) for example.
+
+I would also suggest using Express as a server for this UI. This is due to it's small footprint, modularity and available 3rd party modules, such as [helmet for http security](https://helmetjs.github.io/), [Passport.js for authentication and authorization support](http://www.passportjs.org/) or [the graphql and express-graphql modules for GraphQL server support](https://graphql.org/graphql-js/running-an-express-graphql-server/).
+
+This UI would also be provided with a full set of supporting elements - such as end to end tests, automation (examples of which are [in this section](#a-ui-repository-and-ways-of-working-in-the-repository)) to manage common tasks, and documentation around implementation approach, UI best practise and so on.
+
+### A UI repository, and ways of working in the repository
+
+In addition to the implementation detailed above, I would propose that any UI contributed to Strimzi be managed and developed as follows:
+
+- Be developed in a dedicated repository inside the Strimzi Github organisation
+- Both the server and client are contained in this repo - worked as sub modules in a monorepo
```

Comment:
I had same doubt reading this line the first time but I then realize that the admin API is referred as "backend" in the proposal. Anyway making it clearer is for sure better.

Reply:
> I assume this refers to the Espress based server and not to the APIs used by the UI/UI backend which will be in the `strimzi-api` project, right? It might be useful to explain it more explicitely.

It does - I will update the wording, add add a diagram to illustrate the proposed topology